### DISCUSSION TITLE
How to override terminal colors?

### DISCUSSION THREAD

**Discussion by pkazmier** [catppuccin/nvim#674] [Q&A] [answered]
I'm trying to figure out how to override some of the terminal colors without changing the color palette. I was hoping to find a `terminal_overrides` option, so I could do something like:

```lua
terminal_overrides = function(C)
    return {
        terminal_color_6 = C.mauve,
        terminal_color_14 = C.mauve,
      }
    end
```

I want to swap out `sky` for `mauve`, but I don't want to change the color palette as it looks great everywhere else. This is for a custom variant https://github.com/catppuccin/nvim/discussions/323#discussioncomment-8653291 I'm building for myself. Thanks!

**Comment by mrtnvgr** [comment]
Changing terminal colors in neovim won't change them in your terminal emulator. If you want to change your terminal palette, you should modify settings of your terminal application.

**Reply by pkazmier** [reply]
When I open a terminal inside of neovim, catppuccin allows me to enable overriding terminal colors with its own palette via `term_colors = true`. So I’m trying to find a way to override what colors it uses by default.

**Comment by pkazmier** [comment, marked as answer]
I've figured this out. Will submit a small PR to allow users to override the terminal colors that catppuccin assigns:

```diff
iff --git a/lua/catppuccin/lib/mapper.lua b/lua/catppuccin/lib/mapper.lua
index 2f60323..1d5e436 100644
--- a/lua/catppuccin/lib/mapper.lua
+++ b/lua/catppuccin/lib/mapper.lua
@@ -56,6 +56,16 @@ function M.apply(flavour)

    theme.integrations = final_integrations -- plugins
    theme.terminal = require("catppuccin.groups.terminal").get() -- terminal colors
+
+   local user_terminal = O.terminal_overrides
+   if type(user_terminal[flavour]) == "function" then user_terminal[flavour] = user_terminal[flavour](C) end
+   theme.terminal = vim.tbl_deep_extend(
+       "keep",
+       user_terminal[flavour] or {},
+       type(user_terminal.all) == "function" and user_terminal.all(C) or user_terminal.all or {},
+       theme.terminal
+   )
+
    local user_highlights = O.highlight_overrides
    if type(user_highlights[flavour]) == "function" then user_highlights[flavour] = user_highlights[flavour](C) end
    theme.custom_highlights = vim.tbl_deep_extend(
```

Then, if one uses `term_colors = true`, they have the ability to override the colors that catppuccin assigns the terminal via:
```lua
    term_colors = true,
    terminal_overrides = {
      all = function(C)
        return {
          terminal_color_3 = C.mauve,
          terminal_color_11 = C.mauve,
        }
      end,
      mocha = function(C)
        return {
          terminal_color_5 = C.pink,
          terminal_color_13 = C.pink,
        }
      end,
    },
```

**Reply by pkazmier** [reply]
Submitted #678 for this feature.
